### PR TITLE
Parenthesize break values containing leading label

### PR DESCRIPTION
**Pull request by dtolnay**
The AST pretty printer previously produced invalid syntax in the case of `break` expressions with a value that begins with a loop or block label.

```rust
macro_rules! expr {
    ($e:expr) => {
        $e
    };
}

fn main() {
    loop {
        break expr!('a: loop { break 'a 1; } + 1);
    };
}
```

`rustc -Zunpretty=expanded main.rs `:

```console
#![feature(prelude_import)]
#![no_std]
#[prelude_import]
use ::std::prelude::rust_2015::*;
#[macro_use]
extern crate std;
macro_rules! expr { ($e:expr) => { $e }; }

fn main() { loop { break 'a: loop { break 'a 1; } + 1; }; }
```

The expanded code is not valid Rust syntax. Printing invalid syntax is bad because it blocks `cargo expand` from being able to format the output as Rust syntax using rustfmt.

```console
error: parentheses are required around this expression to avoid confusion with a labeled break expression
 --> <anon>:9:26
  |
9 | fn main() { loop { break 'a: loop { break 'a 1; } + 1; }; }
  |                          ^^^^^^^^^^^^^^^^^^^^^^^^
  |
help: wrap the expression in parentheses
  |
9 | fn main() { loop { break ('a: loop { break 'a 1; }) + 1; }; }
  |                          +                        +
```

This PR updates the AST pretty-printer to insert parentheses around the value of a `break` expression as required to avoid this edge case.

### CODE REVIEW

File: compiler/rustc_ast/src/util/classify.rs [hunk 1]
```diff
@@ -81,6 +82,82 @@ pub fn expr_requires_semi_to_be_stmt(e: &ast::Expr) -> bool {
...
+            | Try(e) => {
+                expr = e;
+            }
+            MethodCall(method_call) => {
+                expr = &method_call.receiver;
+            }
+
+            AddrOf(..)
+            | Array(..)
+            | Await(..)
```
Comment (fmease) [inline compiler/rustc_ast/src/util/classify.rs:126]: `Await` also belongs to the category above, we need to recurse into it. Consider:

```rs
async fn f() {
    loop {
        break expr!('a: loop { break 'a async { 0 }; }).await;
    };
}
```

**Review by fmease — APPROVED** [review]
Thanks! Sorry for the delay.
r=me with nit addressed

File: compiler/rustc_ast/src/util/classify.rs (hunk 1 above)
Comment (dtolnay) [reply compiler/rustc_ast/src/util/classify.rs]: Sure enough &mdash; good catch.

**Pull request merged**